src/core/Akka/Actor/Deployer.cs

```diff
@@ -100,11 +100,20 @@ private RouterConfig CreateRouterConfig(string routerTypeAlias, Config deploymen
 
             var path = string.Format("akka.actor.router.type-mapping.{0}", routerTypeAlias);
             var routerTypeName = _settings.Config.GetString(path);
-            var routerType = Type.GetType(routerTypeName);
-            Debug.Assert(routerType != null, "routerType != null");
-            var routerConfig = (RouterConfig)Activator.CreateInstance(routerType, deployment);
 
-            return routerConfig;
+            try
+            {
+                var routerType = Type.GetType(routerTypeName, true);
+                Debug.Assert(routerType != null, "routerType != null");
+                var routerConfig = (RouterConfig)Activator.CreateInstance(routerType, deployment);
+
+                return routerConfig;
+            }
+            catch (TypeLoadException ex)
+            {
+                var msgException = string.Format("Could not find type '{0}'", routerTypeName);
```

Comment:
We should indicate that the type is intended for a Rotuer Config.
e.g. "Could not find the router config type ${type}"

we should also throw an `ConfigurationException` if the type is not assignable to `Akka.Routing.RouterConfig`